forem/forem
# Implement Unified Embed for CodeSandbox URLs

## What type of PR is this? (check all applicable)


## Description
This PR implements the [unified embed tag](https://github.com/forem/forem/issues/15099) for CodeSandbox embeds.

## Related Tickets & Documents
closes https://github.com/forem/forem/issues/15554

## QA Instructions, Screenshots, Recordings

Example CodeSandbox URLs:
https://app.codesandbox.io/embed/exciting-knuth-hywlv
https://codesandbox.io/embed/exciting-knuth-hywlv?file=/index.html&runonclick=0&view=editor
Note: to pass in options using the `{% codesandbox %}` embed, pass them in space-seperated like: `file=/index.html runonclick=0 view=editor`

- As a user, create a CodeSandbox liquid tag using this format: `{% embed <codesandbox-url> %}`. Please use the example URLs provided above. The CodeSandbox Liquid Tag should render properly.
- Also create CodeSandbox liquid tag using the old method: `{% codesandbox <codesandbox-id> %}` or `{% codesandbox <codesandbox-id> <space-delimited-options> %}`. This should also work, as I am leaving the old implementation in place for now.

### UI accessibility concerns?

None

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?
Considering the project as a whole, I shall update the Liquid Tag documentation at the end. Since I am not changing the current implementation, the present documentation is still valid. Plus, I will avoid updating the docs piecemeal, and risk confusing users.
